ouroboros-consensus-diffusion/test/consensus-test/Test/Consensus/PeerSimulator/ChainSync.hs

```diff
@@ -0,0 +1,135 @@
+import           Ouroboros.Network.Block (Tip)
+import           Ouroboros.Network.Channel (createConnectedChannels)
+import           Ouroboros.Network.ControlMessage (ControlMessage (..))
+import           Ouroboros.Network.Driver.Limits
+                     (ProtocolLimitFailure (ExceededSizeLimit, ExceededTimeLimit))
+import           Ouroboros.Network.NodeToNode.Version (NodeToNodeVersion)
+import           Ouroboros.Network.Protocol.ChainSync.ClientPipelined
+                     (ChainSyncClientPipelined, chainSyncClientPeerPipelined)
+import           Ouroboros.Network.Protocol.ChainSync.Codec (ChainSyncTimeout,
+                     codecChainSyncId, timeLimitsChainSync)
+import           Ouroboros.Network.Protocol.ChainSync.PipelineDecision
+                     (pipelineDecisionLowHighMark)
+import           Ouroboros.Network.Protocol.ChainSync.Server (ChainSyncServer,
+                     chainSyncServerPeer)
+import           Test.Consensus.Network.Driver.Limits.Extras
+                     (chainSyncNoSizeLimits,
+                     runConnectedPeersPipelinedWithLimits)
+import           Test.Consensus.PeerSimulator.StateView
+                     (ChainSyncException (ChainSyncException),
+                     StateViewTracers (StateViewTracers, svtChainSyncExceptionsTracer))
+import           Test.Consensus.PeerSimulator.Trace (mkChainSyncClientTracer,
+                     traceUnitWith)
+import           Test.Consensus.PointSchedule (PeerId, TestFragH)
+import           Test.Util.Orphans.IOLike ()
+import           Test.Util.TestBlock (TestBlock)
+
+
+basicChainSyncClient :: forall m.
+  IOLike m =>
+  Tracer m String ->
+  TopLevelConfig TestBlock ->
+  ChainDbView m TestBlock ->
+  StrictTVar m TestFragH ->
+  (m (), m ()) ->
+  Consensus ChainSyncClientPipelined TestBlock m
+basicChainSyncClient tracer cfg chainDbView varCandidate (startIdling, stopIdling) =
+  chainSyncClient
+    CSClient.ConfigEnv {
+        CSClient.mkPipelineDecision0     = pipelineDecisionLowHighMark 10 20
+      , CSClient.tracer                  = mkChainSyncClientTracer tracer
+      , CSClient.cfg
+      , CSClient.chainDbView
+      , CSClient.someHeaderInFutureCheck = dummyHeaderInFutureCheck
+      }
+    CSClient.DynamicEnv {
+        CSClient.version             = maxBound
+      , CSClient.controlMessageSTM   = return Continue
+      , CSClient.headerMetricsTracer = nullTracer
+      , CSClient.varCandidate
+      , CSClient.startIdling
+      , CSClient.stopIdling
+      }
+  where
+    dummyHeaderInFutureCheck ::
+      InFutureCheck.SomeHeaderInFutureCheck m TestBlock
+    dummyHeaderInFutureCheck =
+      InFutureCheck.SomeHeaderInFutureCheck InFutureCheck.HeaderInFutureCheck
+      { InFutureCheck.proxyArrival = Proxy
+      , InFutureCheck.recordHeaderArrival = \_ -> pure ()
+      , InFutureCheck.judgeHeaderArrival = \_ _ _ -> pure ()
+      , InFutureCheck.handleHeaderArrival = \_ -> pure Nothing
+      }
+
+runChainSyncClient ::
```

Comment:
What do you think of a35d82e318e246937841dff03f5396fffa820524? I'm not really so keen on bundling the arguments in a datatype, but that is mostly because I know it will cause many conflicts with subsequent PRs. Other than that, it sounds like a pretty good idea tbh.